File: Server/models/Wishlist.js
const mongoose = require('mongoose');
const Counter = require('./Counter');

const wishlistSchema = new mongoose.Schema({
  wishlist_id: {
    type: Number,
    unique: true
  },
  user_id: {
    type: Number,
    required: true,
    ref: 'User',
    index: true
  },
  template_id: {
    type: Number,
    ref: 'SystemClothing',
    default: null,
    index: true
  },
  item_name: {
    type: String,
    required: true,
    trim: true
  },
  image_url: {
    type: String,
    default: null
  },
  price_estimate: {
    type: Number, 
    default: null,
    min: 0
  },
  link_store: {
    type: String,
    default: null,
    trim: true
  },
  status: {
    type: String,
    enum: ['pending', 'purchased'], 
    default: 'pending', 
    index: true 
  }
}, {
  timestamps: true
});

wishlistSchema.pre('save', async function() {
  const doc = this;

  if (!doc.isNew) {
    return;
  }

  const counter = await Counter.findByIdAndUpdate(
    { _id: 'wishlist_id' },
    { $inc: { seq: 1 } },
    { new: true, upsert: true }
  );

  doc.wishlist_id = counter.seq;
});

module.exports = mongoose.model('Wishlist', wishlistSchema, 'wishlists');

File: Android/app/src/main/java/com/example/smartfashion/model/Wishlist.kt
package com.example.smartfashion.model

import com.google.gson.annotations.SerializedName
import java.util.Date

data class Wishlist(
    @SerializedName("wishlist_id")
    val wishlistId: Int? = null,

    @SerializedName("user_id")
    val userId: Int,

    @SerializedName("item_name")
    val itemName: String,

    @SerializedName("image_url")
    val imageUrl: String? = null,

    @SerializedName("price_estimate")
    val priceEstimate: Double? = null,

    @SerializedName("link_store")
    val linkStore: String? = null,

    @SerializedName("is_favorite")
    val isFavorite: Boolean = false,

    @SerializedName("status")
    val status: String = "pending",

    @SerializedName("createdAt")
    val createdAt: Date? = null,

    @SerializedName("updatedAt")
    val updatedAt: Date? = null
)